packages/log-view/src/browser/log-model.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { Emitter, Event } from "@theia/core";
+import { LogMessageParams } from "vscode-languageserver-protocol";
+import { VirtualElement, h } from "@phosphor/virtualdom";
+import { MessageType } from "@theia/core";
+import { injectable } from "inversify";
+
+export const ALL_CHANNEL_NAME = 'All';
+
+@injectable()
+export class LogModel {
+
+    private channels = new Map<string, LogChannel>();
+    private contentChangeEmitter = new Emitter<LogChannel>();
+    private channelsChangedEmitter = new Emitter<LogChannel[]>();
+    private allChannel: LogChannel;
+
+    onContentChange: Event<LogChannel> = this.contentChangeEmitter.event;
+    onChannelsChange: Event<LogChannel[]> = this.channelsChangedEmitter.event;
+
+    constructor() {
+        this.allChannel = this.getChannel(ALL_CHANNEL_NAME);
+    }
+
+    getChannel(channelName: string): LogChannel {
+        let channel = this.channels.get(channelName);
+        if (!channel) {
+            channel = new LogChannel(channelName, this);
+            this.channels.set(channelName, channel);
+            this.channelsChangedEmitter.fire(Array.from(this.channels.values()));
+        }
+        return channel;
+    }
+
+    getChannelNames(): string[] {
+        return Array.from(this.channels.keys());
+    }
+
+    fireLogChannelChanged(logChannel: LogChannel, message: LogMessageParams) {
```

Comment:
should it be exposed to clients?

Reply:
Nope. What should I do to allow to communicate between LogChannel and LogModel? Use an event instead, introduce an interface, or merge LogModel and LogChannel?